bash/code_to_gast/bash_helpers.js

```diff
@@ -0,0 +1,46 @@
+const bash_router = require("./bash_router.js")
+
+
+function script_to_gast(node) {
+    gast = {"type": "root"}
+    gast["body"] =  bash_router.node_to_gast(node["commands"])
+    return gast
+}
+
+function array_to_gast(node) {
+    let node_list = []
+    for(const elem of node) {
+        node_list.push(bash_router.node_to_gast(elem))
+    }
+    return node_list
+}
+
+function word_to_gast(node) {
+    text = node["text"]
+    if (text === "true") {
+        return {"type": "bool", "value": 1}
+    } else if (text === "false") {
+        return {"type": "bool", "value": 0}
+    } else if (!isNaN(text)) {
```

Comment:
this seems dangerous... maybe add comment explaining what is happening here so that it is easier to fix if it breaks

Reply:
Okay

Reply:
I had no idea `isNaN` has some fancy stuff when passed a string ...